finality-aleph/src/network/mock.rs

```diff
@@ -10,14 +13,30 @@ use std::{
     sync::Arc,
 };
 
-type Channel<T> = (
-    Arc<Mutex<mpsc::UnboundedSender<T>>>,
-    Arc<Mutex<mpsc::UnboundedReceiver<T>>>,
+#[derive(Clone)]
+pub struct Channel<T>(
+    pub mpsc::UnboundedSender<T>,
+    pub Arc<tokio::sync::Mutex<mpsc::UnboundedReceiver<T>>>,
```

Comment:
Huh, why is the other mutex still here? Could you remove it?

Reply:
The methods here are not async (in trait that is being mocked) but still need mutexes in tests.